src/main/java/com/bookstore/bookstoreapi/data/entity/Wishlist.java

```diff
@@ -40,5 +40,5 @@ public class Wishlist {
   @ToString.Exclude
   @Builder.Default
   @ManyToMany(mappedBy = "wishlists")
-  private List<Book> books = new ArrayList<>();
+  private Set<Book> books = new HashSet<>();
```

Comment:
This is another safeguard that I thought would be nice, since sets are non-repeatable, meaning cannot have the same item twice, it's perfect for this situation.